src/Data/Nat/Binary/Minus.agda

```diff
@@ -0,0 +1,364 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Subtraction on Bin and some of its property proofs.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Nat.Binary.Minus where
+
+open import Algebra using (Magma)
+import Algebra.FunctionProperties as FuncProp
+open import Data.Empty using (⊥)
+open import Data.Fin using (#_)
+open import Data.Nat as ℕ using (ℕ)
+open import Data.Nat.Binary.Base
+open import Data.Nat.Binary.Properties
+import Data.Nat.Properties as ℕₚ
+open import Data.Product using (_×_; _,_; proj₁; proj₂; ∃)
+open import Data.Sum using (inj₁; inj₂)
+open import Data.Vec using ([]; _∷_)
+open import Function using (_∘_; _$_)
+open import Level using (0ℓ)
+open import Relation.Binary using (Tri; _Preserves_⟶_; _Preserves₂_⟶_⟶_)
+open import Relation.Binary.PropositionalEquality as P using
+  (_≡_; _≢_; _≗_; refl; sym; trans; cong; cong₂; resp₂; subst; magma;
+   module ≡-Reasoning)
+open import Relation.Nullary using (Dec; yes; no)
+open import Relation.Nullary.Negation using (contradiction)
+open import Algebra.FunctionProperties {A = ℕᵇ} _≡_
+open import Algebra.Properties.CommutativeSemigroup +-commutativeSemigroup
+  using (xy∙z≈y∙xz; x∙yz≈y∙xz)
+open import Algebra.Solver.CommutativeMonoid +-0-commutativeMonoid
+
+private module FP-Bin = FuncProp (_≡_ {A = ℕᵇ})
+open Tri
+
+infixl 6 _∸_
+
+_∸_ : Op₂ ℕᵇ
+zero     ∸ _        = 0ᵇ
+x        ∸ zero     = x
+2[1+ x ] ∸ 2[1+ y ] = double (x ∸ y)
+1+[2 x ] ∸ 1+[2 y ] = double (x ∸ y)
+2[1+ x ] ∸ 1+[2 y ] with x <? y
+... | yes _ = 0ᵇ
+... | no _  = 1+[2 (x ∸ y) ]
+1+[2 x ] ∸ 2[1+ y ] with x ≤? y
+... | yes _ = 0ᵇ
+... | no _  = pred (double (x ∸ y))
+
+x∸0≡x : ∀ x → x ∸ 0ᵇ ≡ x
```

Comment:
Please could you have a look at `Data.Nat.Properties` and add sections in the same style?

Reply:
Done.